Objects/unicodeobject.c

```diff
@@ -15819,7 +15852,12 @@ unicode_iter(PyObject *seq)
     }
     if (PyUnicode_READY(seq) == -1)
         return NULL;
-    it = PyObject_GC_New(unicodeiterobject, &PyUnicodeIter_Type);
+    if (PyUnicode_IS_COMPACT_ASCII(seq)) {
+        it = PyObject_GC_New(unicodeiterobject, &_PyUnicodeASCIIIter_Type);
```

Comment:
Is it possible for a `unicodeiterobject` to participate in a reference cycle? Could we get rid of the GC headers?

Reply:
Ah yes, str subclasses can have cyclic references.